poutyne/framework/callbacks/tracker.py

```diff
@@ -0,0 +1,176 @@
+from typing import Dict, List, Tuple
+
+import numpy as np
+import torch
+from torch.utils.tensorboard import SummaryWriter
+
+from .callbacks import Callback
+
+
+class Tracker(Callback):
+    def __init__(self, keep_bias: bool = False) -> None:
+        super().__init__()
+
+        self.keep_bias = keep_bias
+        self.writer = None
+
+        self.layer_names = []
+        self.number_layers = 0
+
+        self.tracker = None
+
+    def on_train_batch_end(self, batch: int, logs: Dict):
+        # pylint: disable=unused-argument
+        named_parameters = ((n, p) for n, p in self.model.network.named_parameters() if self._keep_layer(p, n))
+        self.tracker.batch_statistic_upgrade(named_parameters)
+
+    def on_train_begin(self, logs: Dict):
+        for layer_name, layer_params in self.model.network.named_parameters():
+            self._get_layers_to_track(layer_name, layer_params)
+        self.tracker = WeightsGradientsStatsTracker(self.number_layers)
+
+    def on_train_end(self, logs: Dict):
+        self.writer.close()
+
+    def on_epoch_end(self, epoch: int, logs: Dict):
+        self._on_epoch_end_log(epoch, logs)
+
+    def _on_epoch_end_log(self, epoch: int, logs: Dict):
+        """
+        The method to define the behavior of the logging tracker into a writer.
+
+        Args:
+            epoch (int): The epoch number.
+            logs (Dict): The epoch logs dictionary.
+        """
+        pass
+
+    def _get_layers_to_track(self, layer_name: str, layer_params: torch.nn.parameter.Parameter):
```

Comment:
The name of this method should not start with get if it does not return anything.

Reply:
true. Better 06bea3f ?